app/controllers/application_controller.rb

```diff
@@ -1,2 +1,26 @@
 class ApplicationController < ActionController::Base
+    # skip_before_action :verify_authenticity_token
+
+    def authenticate_user_using_x_auth_token
+        user_email = request.headers["X-Auth-Email"]
+        auth_token = request.headers["X-Auth-Token"].presence
+        user = user_email && User.find_by_email(user_email)
+    
+        if user && auth_token &&
+          ActiveSupport::SecurityUtils.secure_compare(
+            user.authentication_token, auth_token
+          )
+          @current_user = user
+        else
+          render status: :unauthorized, json: {
+            errors: ["Could not authenticate with the provided credentials"]
+          }
+        end
+      end
+    
+      private
```

Comment:
Unused private method

Reply:
Actually, I am using this method in polls_controller [in line number 10]

I just found that in ruby we can call a private method from within the class it is declared in as well as all subclasses of that class. We can declare a private method in ApplicationController and have it available in all other controllers.

Reply:
Yes. This was a misunderstanding on my part, too. In Java, private methods cannot be inherited amd I wrongly assumed the same for Ruby. Thanks for pointing this out.